fe/fe-common/src/main/java/org/apache/doris/catalog/ArrayType.java

```diff
@@ -88,13 +88,11 @@ public boolean matchesType(Type t) {
             return false;
         }
 
-        // Array(Null) is a virtual Array type, can match any Array(...) type
-        if (itemType.isNull() || ((ArrayType) t).getItemType().isNull()) {
-            return true;
+        if (((ArrayType) t).getContainsNull() != getContainsNull()) {
```

Comment:
what's the sematics of getContainsNull() in ArrayType?

Reply:
What's the difference to itemType.isNull() ?

Reply:
getContainsNull()  means nullable(Array)
getItemType().isNull() means current item is null_type